components/leave_private_channel_modal/leave_private_channel_modal.tsx

```diff
@@ -5,23 +5,53 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import {FormattedMessage} from 'react-intl';
 
+import {Channel} from 'mattermost-redux/types/channels';
+
 import ModalStore from 'stores/modal_store.jsx';
 import Constants from 'utils/constants';
 import ConfirmModal from 'components/confirm_modal.jsx';
 
-export default class LeavePrivateChannelModal extends React.Component {
+type State = {
+    show: boolean;
+    channel: Channel;
+};
+
+type Props = {
+    actions: {
+        leaveChannel: (channelId: any) => any;
+    };
+}
+
+export default class LeavePrivateChannelModal extends React.Component<Props, State> {
     static propTypes = {
         actions: PropTypes.shape({
             leaveChannel: PropTypes.func.isRequired,
         }).isRequired,
     };
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
 
         this.state = {
             show: false,
-            channel: null,
+            channel: {
```

Comment:
I think this is not what is expected, I would expect here to have a null value, not a fake null channel. If this state is null, the component itself should behave as expected, there is no point to show the modal at all if we don't have a channel to leave, so I would check if the channel exist, and if not, just don't show the modal.

Reply:
@jespino, since the type of channel is `Channel` i am receiving an error when i try to assign a null value to it! `Type 'null' is not assignable to type 'Channel'.`

Reply:
Yes, then you have to options, allow to store null or channel in that field (`channel: null | Channel`), or add that field as optional like `channel?: Channel` and then instead of using `null` use the idea of not passing the value until is defined.

Reply:
I prefer the later option when is possible, because is more explicit about "this is undefined until is defined, and after that, the normal expected behavior is, is defined always", if that is the case (which I think it is in this case) I think is a better approach to make it optional.

Reply:
@jespino Thank you for helping me out!

Reply:
@sowmiyamuthuraman Thank **you** for helping us out!